app/controllers/StoryPackagesController.scala

```diff
@@ -1,14 +1,14 @@
 package controllers
 
-import java.net.URLDecoder
+import java.net.URLEncoder
```

Comment:
I just realized now (looking at this change) that the query comes already encoded from the client. If we encode again we end up encoding twice, which is bad.

Not sure what's the solution.
Not encoding means trusting the client is doing the encoding, if we don't trust the client then maybe we should decode first and encode it again.